Makefile

```diff
@@ -106,7 +115,7 @@ build: ## build binaries
 .PHONY: test
 test: ## Run tests
 	@echo -e "\033[32mTesting...\033[0m"
-	$(DOCKER_CMD) hack/ci-test.sh
+	$(DOCKER_CMD) KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path --bin-dir $(PROJECT_DIR)/bin)" hack/ci-test.sh
```

Comment:
when i run `NO_DOCKER=0 make test` this command fails because of the environment variable assignment. error looks like this:

```
$ make test
Testing...
podman  run --rm -e CGO_ENABLED=1 -e GOARCH=amd64 -e GOOS=linux -v "/home/mike/dev/machine-api-provider-azure":/go/src/github.com/openshift/machine-api-provider-azure:Z -w /go/src/github.com/openshift/machine-api-provider-azure registry.ci.openshift.org/openshift/release:golang-1.18 KUBEBUILDER_ASSETS="/home/mike/dev/machine-api-provider-azure/bin/k8s/1.25.0-linux-amd64" hack/ci-test.sh
Error: executable file `KUBEBUILDER_ASSETS=/home/mike/dev/machine-api-provider-azure/bin/k8s/1.25.0-linux-amd64` not found in $PATH: No such file or directory: OCI runtime attempted to invoke a command that was not found
make: *** [Makefile:118: test] Error 127
```

i think we need to embed the environment variable as a `--env` option to the container runtime.